src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1071,5 +1071,42 @@ static internal Version GetAssemblyVersion()
 
             return s_systemDataVersion;
         }
+
+
+        static internal readonly string[] AzureSqlServerEndpoints = {Res.GetString(Res.AZURESQL_GenericEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_GermanEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_UsGovEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_ChinaEndpoint) };
+
+        static internal bool IsAzureSqlServerEndpoint(string dataSource)
+        {
+            // remove server port
+            var i = dataSource.LastIndexOf(',');
+            if (i >= 0)
+            {
+                dataSource = dataSource.Substring(0, i);
+            }
+
+            // check for the instance name
+            i = dataSource.LastIndexOf('\\');
+            if (i >= 0)
+            {
+                dataSource = dataSource.Substring(0, i);
+            }
+
+            // trim redundant whitespaces
+            dataSource = dataSource.Trim();
+
+            // check if servername end with any azure endpoints
+            for (i = 0; i < AzureSqlServerEndpoints.Length; i++)
```

Comment:
Could probably use "foreach(string endpoint in AzureSqlServerEndpoints)" here, as opposed to manual indexing.